include/pinocchio/spatial/inertia.hpp

```diff
@@ -889,6 +960,366 @@ namespace pinocchio
 
   }; // class InertiaTpl
 
+  /**
+   * @brief A structure representing a pseudo inertia matrix.
+   *
+   * References:
+   * - Wensing, Patrick M., Sangbae Kim, and Jean-Jacques E. Slotine. "Linear matrix inequalities
+   * for physically consistent inertial parameter identification: A statistical perspective on the
+   * mass distribution." IEEE Robotics and Automation Letters 3.1 (2017): 60-67.
+   */
+  template<typename Scalar, int Options>
+  struct PseudoInertiaTpl
+  {
+    typedef Eigen::Matrix<Scalar, 4, 4, Options> Matrix4;
+    typedef Eigen::Matrix<Scalar, 3, 1, Options> Vector3;
+    typedef Eigen::Matrix<Scalar, 3, 3, Options> Matrix3;
+    typedef Eigen::Matrix<Scalar, 10, 1, Options> Vector10;
+    typedef LogCholeskyParametersTpl<Scalar, Options> LogCholeskyParameters;
+
+    Scalar mass;    ///< Mass of the pseudo inertia
+    Vector3 h;      ///< Vector part of the pseudo inertia
+    Matrix3 Sigma;  ///< 3x3 matrix part of the pseudo inertia
+    Matrix4 matrix; ///< Full 4x4 pseudo inertia matrix
+
+    PseudoInertiaTpl(Scalar mass, const Vector3 & h, const Matrix3 & Sigma)
+    : mass(mass)
+    , h(h)
+    , Sigma(Sigma)
+    {
+      matrix = toMatrix();
+    }
+
+    explicit PseudoInertiaTpl(const Matrix4 & pseudo_inertia_matrix)
+    {
+      *this = FromMatrix(pseudo_inertia_matrix);
+    }
+
+    explicit PseudoInertiaTpl(const InertiaTpl<Scalar, Options> & inertia)
+    {
+      *this = FromInertia(inertia);
+    }
+
+    explicit PseudoInertiaTpl(const Vector10 & dynamic_params)
+    {
+      *this = FromDynamicParameters(dynamic_params);
+    }
+
+    explicit PseudoInertiaTpl(const LogCholeskyParameters & log_cholesky)
+    {
+      *this = log_cholesky.toPseudoInertia();
+    }
```

Comment:
In think you can remove these constructors.

They are redundant with the static functions FromInertia, FromDynamicParameters, FromMatrix.

I prefer having on constructor allowing to initialize easily all the class parameters (like this one `PseudoInertiaTpl(Scalar mass, const Vector3 & h, const Matrix3 & Sigma)`) and static function as conversion/helper constructors.

This is more explicit and avoid overloading issues (what can we do if there is two way to create a PseudInertia from a Vector10 ?).